File: Java/DeepSpace/src/deepspace/WeaponType.java
package deepspace;

/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */

/**
 *
 * @author victor
 */
public enum WeaponType {
    LASER(2.0f), MISSILE(3.0f), PLASMA(4.0f);  
    
    private final float power;
    
    WeaponType(float power) {
        this.power = power;
    }
    
    float getPower() {
        return power;
    }  

    @Override
    public String toString() {
        String res;
        switch (this) {
            case LASER:
                res = "Laser{" + "power=" + power + '}'; 
                break;
            case MISSILE:
                res = "Missile{" + "power=" + power + '}'; 
                break;
            case PLASMA:
                res = "Plasma{" + "power=" + power + '}';
                break;
            default:
                res = "Error";
                break;
        }
        return res;
    }   
    
}

File: Deepspace_ruby/Deepspace/lib/WeaponType.rb
#encoding:utf-8
module Deepspace
module WeaponType
    class Type
      attr_reader :power, :name
      
      def initialize(pow, s)
        @power = pow
        @name = s
      end
      
      def to_s
        "#{@name}, Power #{@power}"
      end
      
    end
    
    LASER = Type.new(2.0, "LASER")
    MISSILE = Type.new(3.0, "MISSILE")
    PLASMA = Type.new(4.0, "PLASMA")
    
    
end
end